src/everest/optimizer/everest2ropt.py

```diff
@@ -107,22 +107,21 @@ def _parse_objectives(objective_functions: list[ObjectiveFunctionConfig], ropt_c
 def _parse_input_constraints(
     controls: FlattenedControls,
     input_constraints: list[InputConstraintConfig] | None,
+    formatted_control_names: list[str],
+    formatted_control_names_dotdash: list[str],
     ropt_config,
 ):
     if not input_constraints:
         return
 
-    # TODO: Issue #9816 is intended to address the need for a more general
-    # naming scheme. This code should be revisited once that issue is resolved.
-    # Ideally the formatted_control_names property of the config is used.
-    formatted_names = [
-        (
-            f"{control_name[0]}.{control_name[1]}-{control_name[2]}"
-            if len(control_name) > 2
-            else f"{control_name[0]}.{control_name[1]}"
-        )
-        for control_name in controls.names
-    ]
+    def _get_control_index(name: str):
+        try:
+            matching_index = formatted_control_names.index(name.replace("-", "."))
+            return matching_index
+        except ValueError:
+            pass
+
+        return formatted_control_names_dotdash.index(name)
```

Comment:
Eventually this should disappear, since using the dash is deprecated. Maybe a comment?